src/DataDictionary/graph/GraphDrawer/GraphDrawer.jsx

```diff
@@ -0,0 +1,249 @@
+import './GraphDrawer.css';
+
+class GraphDrawer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.currentHighlightingNodeClassName = 'graph-drawer__node--current-highlighting';
+    this.graphDomRef = React.createRef();
+  }
+
+  componentDidUpdate() {
+    if (this.props.isGraphView
+      && this.props.highlightingNode
+      && !this.props.highlightingNodeSVGElement) {
+      const highlightingNodeSVGElement = this.getHighlightingNodeSVGElement();
+      this.props.onHighlightingNodeSVGElementUpdated(highlightingNodeSVGElement);
+    }
+  }
+
+  onMouseOver = (node, e) => {
+    const hoveringNodeSVGElement = e.currentTarget;
+    this.props.onHoverNode(node, hoveringNodeSVGElement);
+  }
+
+  onMouseOut = () => {
+    this.props.onCancelHoverNode();
+  }
+
+  onClick = (node, e) => {
+    if (!this.props.highlightingNode) { // if no node is highlighted yet
+      const highlightingNodeSVGElement = e.currentTarget;
+      this.props.onClickNode(node, highlightingNodeSVGElement);
+    } else if (this.props.clickableHighlightedNodeIDs.length > 1
+      && this.props.clickableHighlightedNodeIDs.includes(node.id)) {
+      // only allow clicking clickable nodes
+      // (only children of highlighted nodes are designed to be clickable)
+      this.props.onFurtherClickNode(node.id);
+    }
+  }
+
+  getHighlightingNodeSVGElement() {
+    const highlightingNodeSVGElement = document.querySelector(
+      `.${this.currentHighlightingNodeClassName}`,
+    );
+    return highlightingNodeSVGElement;
+  }
+
+  render() {
+    if (!this.props.layoutInitialized) return (<React.Fragment />);
+    const boundingBoxLength = this.props.graphBoundingBox[2][0];
+    const fittingScale = Math.min(
+      this.props.canvasWidth,
+      this.props.canvasHeight,
+    ) / boundingBoxLength;
+    const fittingTransX = Math.abs(
+      (boundingBoxLength - (this.props.canvasWidth / fittingScale)) / 2,
+    );
+    const fittingTransY = Math.abs(
+      (boundingBoxLength - (this.props.canvasHeight / fittingScale)) / 2,
+    );
+    const clickableNodeClassName = 'graph-drawer__node--clickable';
+    const notClickableNodeClassName = 'graph-drawer__node--not-clickable';
+    const highlightedNodeClassName = 'graph-drawer__node--highlighted';
+    const notHighlightedNodeClassName = 'graph-drawer__node--not-highlighted';
+    const highlightedLinkClassName = 'graph-drawer__link--highlighted';
+    const notHighlightedLinkClassName = 'graph-drawer__link--not-highlighted';
+    const requiredLinkClassName = 'graph-drawer__link--required';
+    return (
+      <g
+        className='graph-drawer'
+        transform={`scale(${fittingScale}) translate(${fittingTransX}, ${fittingTransY}) `}
+        ref={this.graphDomRef}
+      >
+        { // edges
+          this.props.edges.map((edge, i) => {
+            let edgeRelatedClassModifier = '';
+            if (this.props.highlightingNode) {
```

Comment:
You should split this function to multiple ones to be more readable.
- draw edges (in some condition)
- draw nodes (in some condition)